Farfetch/blackout
# fix: add consistency fixes

## Description

This adds a set of consistency fixes to some inconsistencies that were
detected throughout a few modules inside all packages and other fixes:

- Renamed `useOrders` and `useReturns` hooks to `useUserOrders`
and `useUserReturns` respectively to be consistent with the
names of other hooks that provide similar logic.
- Changed the order of parameters in actions that allow
overriding the hook's parameters to the last position
as overriding them is not a common usage and being in
the last position makes the methods easier to use for the much more
common usages of not overriding the ones from the hook. This was
changed in hooks `useReturnPickupCapability`,
`useReturnPickupRescheduleRequest`,
`useReturnPickupRescheduleRequests`, `useReturn`, `useOrder`,
`useOrderReturnOptions` and `useOrderReturns`.
- Renamed `useOrder`'s `resetOrderDetailsState` action to `reset`
for consistency with other hooks.
- Changed all hooks' auto fetch condition to use `isFetched` instead of
looking at the error.
- Changed all hooks that used custom logic to calculate if the current
user is authenticated or not to use the `isAuthenticated` user'
selector instead.
- Changed useBag's `fetch` action to implicitly use the bagId passed
when invoking the hook instead of requiring the consumer to
specify it again.
- Changed useBagItem implementation to use a new selector,
isBagItemFetched to derive the information if a bag item is fetched
or not to be consistent with other hooks.
- Changed useUser's `changePassword` and `update` actions to implicitly
use the `userId` that is set in the store instead of requiring the
consumer to specify it.
- Changed useWishlist's fetch action to implicitly use the
wishlistId that was passed to the hook instead of requiring the
consumer to specify it.
- Changed useWishlistItem's `update` and `remove` actions to implicitly
use the `wishlistItemId` that was passed to the hook
instead of requiring the consumer to specify it. Also added a
`isWishlistItemFetched` selector to derive the value of the
`isFetched` property.
- Changed `useContentPage` hook and `fetchContentPage` action to
discard the query string for the hash generation.
- Changed the implementation of the hooks `useWishlistSets`
and `useWishlistSet` to be consistent with the remaining hooks.
- Changed the `bags` and `wishlist` reducers to reset its state
when a fetch user request is performed and returns
different bag or wishlist ids.
- Unified the function signatures of the putUserDefault{Billing,Shipping,
Contact}Address clients.
- Changed the imports that were used in @example tags inside tsdoc to
point to the root of the package (when necessary).
- Moved some of the `useReturn`'s actions to `useUserReturns` for consistency.
- Added a new order status to the OrderStatus enum.
- Renamed type `BreadCrumb` to `ProductsBreadcrumb` to avoid ambiguity
with the existing type `Breadcrumb` from contents area.
- Removed type `PasswordValidationErrorsConstants` and replaced it with
`PasswordValidationErrors` which is now an enum.

BREAKING CHANGE:
- `useOrders` and `useReturns` were renamed to
`useUserOrders` and `useUserReturns` respectively.
- `useBag`, `useBagItem`, `useWishlist`, `useWishlistItem`, `useUser`,
`useWishlistSet`, `useWishlistSets` `useReturnPickupCapability`,
`useReturnPickupRescheduleRequest`,
`useReturnPickupRescheduleRequests`, `useReturn`, `useOrder`,
`useOrderReturnOptions` and `useOrderReturns` have some changes in its
public properties/actions. Check the description for more information.
- `putUserDefaultShippingAddress` and `putUserDefaultBillingAddress`
signatures changed. Now, instead of receiving an object containing the
userId and id properties, there is a specific parameter for both of
those properties.
- `fetchContentPage` action and `useContentPage` hook now discard the
query string from the `codes` in query when generating the hash.
- Type `BreadCrumb` used in product listings was renamed to
`ProductsBreadcrumb`. Do not confuse with the type `Breadcrumb` (with
lowercase c) which is to be used with contents modules.
- Type `PasswordValidationErrorsConstants` was replaced with
`PasswordValidationErrors` type which is now an enum.





### Dependencies



## Checklist



- [X] Tests for the respective changes have been added
- [X] The code is commented, particularly in hard-to-understand areas
- [X] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)
